Added OSX support, but checksum mismatch

Hi,

I added OSX support to modem.sh.

I searched and replaced sha256sum to $sha256sum

In line 12 I added:

if test `uname` == Darwin ; then sha256sum=gsha256sum ; else sha256sum=sha256sum; fi

To use, one needs to install GNU coreutils. Using Homebrew: $ brew install coreutils

Result:

$ gsha256sum modem-17.12.1.zip
d89abda5950d5feda0474a6b4d0122a514da4c58cc212a55ec009666863293ba  modem-17.12.1.zip

On WSL (Windows Subsystem for Linux) it worked out of the box with Ubuntu. However I got the following result:

$ sha256sum modem-17.12.1.zip
dfeb89ed50371b768f2b656d97c4a489dedb7631ef56a6af4484e2097aac2d36  modem-17.12.1.zip

$ make release
mktemp: illegal option -- -
usage: mktemp [-d] [-q] [-t prefix] [-u] template ...
       mktemp [-d] [-q] [-u] -t prefix 
Downloading fp2-sibon-18.04.1-ota-userdebug.zip...
######################################################################## 100.0%
/bin/bash: check: command not found
Unpacking firmware files...
unzip:  cannot find or open ./updates/fp2-sibon-18.04.1-ota-userdebug.zip, ./updates/fp2-sibon-18.04.1-ota-userdebug.zip.zip or ./updates/fp2-sibon-18.04.1-ota-userdebug.zip.ZIP.
make: *** [src/firmware-update] Error 9

--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,19 @@
 # Makefile by Roberto M.F. (Roboe)
 # https://github.com/WeAreFairphone/flashabe-zip_emojione
 
-SHELL     := /bin/bash
+SHELL       := /bin/bash
+
+# Dependencies
+CURL        := $(shell command -v curl 2>&1)
+ZIP         := $(shell command -v zip 2>&1)
+UNZIP       := $(shell command -v unzip 2>&1)
+ifeq ($(uname -s),"Darwin")
+  SHA256SUM := $(shell command -v gsha256sum 2>&1)
+  MKTEMP    := $(shell command -v gmktemp 2>&1)
+else
+  SHA256SUM := $(shell command -v sha256sum 2>&1)
+  MKTEMP    := $(shell command -v mktemp 2>&1)
+endif
 
 # Version and release
 VERSION      := 18.04.1
@@ -15,19 +27,13 @@ SOURCE       := ./src/
 FIRMWARE_DIR := ./src/firmware-update/
 EDIFY_BINARY := ./src/META-INF/com/google/android/update-binary
 EDIFY_SCRIPT := ./src/META-INF/com/google/android/updater-script
-TEMP_DIR     := $(shell mktemp --dry-run -d /tmp/modem.XXXXXXXX)
+TEMP_DIR     := $(shell $(MKTEMP) --dry-run -d /tmp/modem.XXXXXXXX)
 
 # Update ZIPs
 OTA_FILENAME := fp2-sibon-$(VERSION)-ota-userdebug.zip
 OTA_FILE     := ./updates/$(OTA_FILENAME)
 OTA_URL      := https://storage.googleapis.com/fairphone-updates/6cb84543-9614-425d-9ab4-9e80baca2b8f/$(OTA_FILENAME)
 OTA_CHECKSUM := 97b39681b773804c8e12177293171698395c43ad9458c7ba823c85c503f00500
-
-# Dependencies
-CURL      := $(shell command -v curl 2>&1)
-ZIP       := $(shell command -v zip 2>&1)
-UNZIP     := $(shell command -v unzip 2>&1)
-SHA256SUM := $(shell if [[ "$(uname -s)" == "Darwin" ]]; then command -v gsha256sum; else command -v sha256sum; fi)
 
 
 .PHONY: all build clean release install

